File: Leetcode/Is Subsequence/IsSubsequence.cpp
class Solution {
public:
    
    //s-> string to be find
    //t-> input string
    bool isSubsequence(string s, string t) {
       
        //we'll basically compare all the alphabets of s and t in order in linear time complexity of O(n)
        int j=0;
        bool res =true;
        for(int i=0;i<t.length();i++){

            //if the characters match then only we'll increment j index.
            if(t[i]==s[j]){
                j++;
            }
        }

        //if j is not reached til the end of string s that would mean, all characters of s are not present in t in order. 
        //Thus there is no subsequence of t equal to s.
        if(j!=s.length()){
            res=false;
        }

        //return the bool value 
        return res;
    }
};

File: Leetcode/Is Subsequence/IsSubsequence.py
class Solution:
    # s-> string to be find
    # t-> input string
    def isSubsequence(self, s: str, t: str) -> bool:
        j = 0
        # return true if s is empty
        if len(s) < 1:
            return True

        # we will compare all the alphabets of s and t in order
        for i in range(len(t)):
            # if the characters match then only we'll increment j index.
            if s[j] == t[i]:
                j += 1

            # if j is not reached til the end of string s that would mean, all characters of s are not present in t or they are not in order.
            # and if j is equal to length of s it is subsequence of t
            if j == len(s):
                return True
        else:
            return False


